### PR TITLE
Allow customzing css classes for the ErrorsViewStackTracePrinter

### PR DESCRIPTION
**Pull request by codeconsole**
```gsp
<g:renderException exception="${exception}" detailsClass="alert alert-danger" stackClass="bg-body-secondary" snippetClass="bg-body-secondary snippet" lineErrorClass="bg-danger" />
```

### CODE REVIEW

File: grails-bootstrap/src/main/groovy/org/grails/exceptions/reporting/DefaultStackTracePrinter.groovy [hunk 1]
```diff
@@ -23,7 +23,7 @@ import org.codehaus.groovy.control.MultipleCompilationErrorsException
  */
 class DefaultStackTracePrinter implements StackTracePrinter {
 
-    String prettyPrint(Throwable t) {
+    String prettyPrint(Throwable t, Map attrs = [:]) {
```
Comment (matrei) [inline grails-bootstrap/src/main/groovy/org/grails/exceptions/reporting/DefaultStackTracePrinter.groovy]: Using Groovy default params on overridden/implemented methods makes the code hard to follow. Can we use:
```groovy
@Override
String prettyPrint(Throwable t) {
    prettyPrint(t, [:])
}

String prettyPrint(Throwable t, Map attrs) {
    
}
```

(or change the interface to include both versions)

File: grails-core/src/main/groovy/org/grails/core/exceptions/DefaultErrorsPrinter.groovy [hunk 2]
```diff
@@ -47,7 +47,7 @@ class DefaultErrorsPrinter extends DefaultStackTracePrinter implements CodeSnipp
         res == null ? te.className : res.getFilename()
     }
 
-    String prettyPrintCodeSnippet(Throwable exception) {
+    String prettyPrintCodeSnippet(Throwable exception, Map attrs = [:]) {
```
Comment (matrei) [inline grails-core/src/main/groovy/org/grails/core/exceptions/DefaultErrorsPrinter.groovy]: Using Groovy default params on overridden/implemented methods makes the code hard to follow. Can we use:
```groovy
@Override
String prettyPrintCodeSnippet(Throwable t) {
    prettyPrintCodeSnippet(t, [:])
}

String prettyPrintCodeSnippet(Throwable t, Map attrs) {
    
}
```

(or change the interface to include both versions)

**Review by matrei — COMMENTED** [review]
You want to add features to the 6.2.x branch?
I'm not sure "this is the way".

**Comment by codeconsole** [comment]
> You want to add features to the 6.2.x branch? I'm not sure "this is the way".

Norgifn breaking here. Does overloading a method really warrant going from 6.2.0 to 6.3.0 if the previous method still exits and works the same as before?

**Comment by matrei** [comment]
> Norgifn breaking here.

You don't know that until it's too late.
Why is this important to get this into Grails 6?

**Comment by codeconsole** [comment]
> You don't know that until it's too late. Why is this important to get this into Grails 6?

It's not critical, but I am trying to overhaul the css in Grails 7 and the only efficient way of me doing it is against live code in Grails 6.   I am trying to stay productive and utilize my time as efficiently as possible. 

All we are doing here is allowing the default css classes to be substituted. Otherwise the same behavior is persisted with no changes.

This is easily tested by just throwing an exception and viewing the error page which I currently am doing locally.
